core/dronecore.cpp

```diff
@@ -23,6 +23,76 @@ DroneCore::~DroneCore()
     _impl = nullptr;
 }
 
+DroneCore::ConnectionResult DroneCore::add_any_connection(std::string str)
+{
+    std::string delimiter = "://";
+    std::string connection_str[3];
+    size_t pos = 0, i = 0;
+    while(i<2)
```

Comment:
This would better be a `for` loop, I think.
And defining `size_t i` with this big scope is not optimal, better just have it inside the loop.